### PR TITLE
Improve ram_group_policy_attachment resource use common method

### PR DESCRIPTION
**Pull request by Lexsss**

### CODE REVIEW

File: alicloud/resource_alicloud_ram_group_policy_attachment.go [hunk 1]
```diff
@@ -38,44 +35,45 @@ func resourceAlicloudRamGroupPolicyAtatchment() *schema.Resource {
...
-	raw, err := client.WithRamClient(func(ramClient *ram.Client) (interface{}, error) {
-		return ramClient.ListPoliciesForGroup(request)
-	})
+	ramService := RamService{client}
+	object, err := ramService.DescribeRamGroupPolicyAttachment(d.Id())
 	if err != nil {
-		if RamEntityNotExist(err) {
-			d.SetId("")
+		if NotFoundError(err) {
 			return nil
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ram_group_policy_attachment.go]: please set id as ""

File: alicloud/resource_alicloud_ram_group_policy_attachment.go [hunk 2]
```diff
@@ -38,44 +35,45 @@ func resourceAlicloudRamGroupPolicyAtatchment() *schema.Resource {
...
 	}
-	response, _ := raw.(*ram.ListPoliciesForGroupResponse)
-	if len(response.Policies.Policy) > 0 {
-		for _, v := range response.Policies.Policy {
+
+	if len(object.Policies.Policy) > 0 {
+		for _, v := range object.Policies.Policy {
 			if v.PolicyName == d.Get("policy_name").(string) && v.PolicyType == d.Get("policy_type").(string) {
-				d.Set("group_name", request.GroupName)
+				d.Set("group_name", d.Get("group_name").(string))
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ram_group_policy_attachment.go]: why change the group_name set?

File: alicloud/resource_alicloud_ram_group_policy_attachment.go [hunk 3]
```diff
@@ -38,44 +35,45 @@ func resourceAlicloudRamGroupPolicyAtatchment() *schema.Resource {
...
-	_, err := client.WithRamClient(func(ramClient *ram.Client) (interface{}, error) {
+	raw, err := client.WithRamClient(func(ramClient *ram.Client) (interface{}, error) {
 		return ramClient.AttachPolicyToGroup(request)
 	})
 	if err != nil {
-		return WrapErrorf(err, DefaultErrorMsg, "ram_group_policy_attachment", request.GetActionName(), AlibabaCloudSdkGoERROR)
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_ram_group_policy_attachment", request.GetActionName(), AlibabaCloudSdkGoERROR)
 	}
+	addDebug(request.GetActionName(), raw)
 	d.SetId("group" + request.PolicyName + string(request.PolicyType) + request.GroupName)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ram_group_policy_attachment.go]: ID？

File: alicloud/service_alicloud_ram.go [hunk 4]
```diff
@@ -247,6 +248,54 @@ func (s *RamService) WaitForRamUser(id string, status Status, timeout int) error
...
+		return ramClient.ListPoliciesForGroup(request)
+	})
+	if err != nil {
+		if RamEntityNotExist(err) {
+			return nil, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		}
+		return nil, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw)
+	response, _ = raw.(*ram.ListPoliciesForGroupResponse)
```
Comment (ddaddaddda) [inline alicloud/service_alicloud_ram.go]: 家类似 if id != id{...}的判断

File: alicloud/service_alicloud_ram.go [hunk 5]
```diff
@@ -247,6 +248,54 @@ func (s *RamService) WaitForRamUser(id string, status Status, timeout int) error
...
+		}
+		if len(object.Policies.Policy) > 0 {
+			if status != Deleted {
+				return nil
+			}
+		} else {
+			if status == Deleted {
+				return nil
+			}
+		}
```
Comment (ddaddaddda) [inline alicloud/service_alicloud_ram.go]: 283-291  我感觉等效于：
if status != Deleted {
	return nil
}

**Pull request merged**